page/pagination.go

```diff
@@ -54,3 +53,26 @@ func (p *Pagination) MutateRequest(req *http.Request) error {
 	}
 	return request.NewParametersMutator(parameters).MutateRequest(req)
 }
+
+type Paginator func(pagination Pagination) (done bool, err error)
+
+func Paginate(paginator Paginator) error {
+	return PaginateWithSize(PaginationSizeDefault, paginator)
+}
+
+func PaginateWithSize(size int, paginator Paginator) error {
+	if size < PaginationSizeMinimum {
+		return errors.New("size is less than minimum")
+	}
+	if paginator == nil {
+		return errors.New("paginator is missing")
+	}
+
+	for page := 0; ; page++ {
```

Comment:
never knew - hadn't spotted that style before

Reply:
Yep, any/all of the three `for` "arguments" are optional.